code/modules/mob/living/simple_animal/hostile/bees.dm

```diff
@@ -82,6 +82,16 @@
 	if(!beehome)
 		. += "<span class='warning'>This bee is homeless!</span>"
 
+/mob/living/simple_animal/hostile/poison/bees/ListTargets() // Bee processing is expessive, so we override them finding targets here.
+	if(!search_objects) //In case we want to have purely hostile bees
+		return ..()
+	else
+		. = list() // The following code is only very slightly slower than just returning oview(vision_range, targets_from), but it saves us much more work down the line
+		var/list/searched_for = oview(vision_range, targets_from)
+		for(var/obj/A in searched_for)
+			. += A
+		for(var/mob/A in searched_for)
```

Comment:
In General, I'm not a big fan of this code, at least as it stands; as it is now, this means the bees almost never attack people unless they don't have a tray to go to; having a bee suit is pretty much unnecessary in botany now because objects are always first in the list.

Reply:
Could you somehow shuffle the list?

Reply:
we don't know how `oview` works, internally, but I suspect it works as a sorting of distance as you describe....but `oview` is what caused "performance" issues to begin with (ok, it wasn't really issues, but someone didn't like how much CPU they were using).

Shuffling the list is possible, but not ideal, I guess.

I'll leave it up to maintainer discretion.

Reply:
You could make it pick() where this list gets used

Reply:
I thought about that, but it's going to likely cause issues: https://github.com/tgstation/tgstation/blob/a148cae769d991e2786ac2277ec979cdf28014a8/code/modules/mob/living/simple_animal/hostile/hostile.dm#L146